### PR TITLE
Handle Circular Reference in MSON Mixins

### PR DESCRIPTION
**Pull request by klokane**
Fix CircularReference with Mixins.

### CODE REVIEW

File: src/refract/ExpandVisitor.cc [hunk 1]
```diff
@@ -194,6 +199,19 @@ namespace refract
...
+                msg <<  "mixin '";
+                msg << href->value;
+                msg << "' circular reference itself";
+
+                throw snowcrash::Error(msg.str(), snowcrash::MSONError);
+            }
+
+            members.push_back(href->value);
+
+
```
Comment (pksunkara) [inline src/refract/ExpandVisitor.cc]: Extra line.

File: src/refract/ExpandVisitor.cc [hunk 2]
```diff
@@ -194,6 +199,19 @@ namespace refract
                 return ref;
             }
 
+            if (std::find(members.begin(), members.end(), href->value) != members.end()) {
+
+                std::stringstream msg;
+                msg <<  "mixin '";
```
Comment (pksunkara) [inline src/refract/ExpandVisitor.cc]: Please correct this to be `named type 'A' is circularly referencing itself`

File: test/fixtures/mson/nested-enum.apib [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# GET /
```
Comment (pksunkara) [inline test/fixtures/mson/nested-enum.apib]: Why is this file here? There is no test case for this.

**Comment by pksunkara** [comment]
@klokane Reviewed. Minor changes to be done. And please correct the situation with circleCI. Apparently, drafter.js tests are failing. /cc @w-vi

File: src/refract/ExpandVisitor.cc (hunk 2 above)
Comment (klokane) [inline src/refract/ExpandVisitor.cc]: I mean, it is good idea to say it is about troubles in mixin.
So I propose something like `named type 'A' is circularly referencing itself by mixin`

File: test/fixtures/mson/nested-enum.apib (hunk 3 above)
Comment (klokane) [inline test/fixtures/mson/nested-enum.apib]: omission, good catch

**Comment by w-vi** [comment]
@klokane drafter.js fixes are in #227 so please merge that one before this PR /cc @pksunkara

File: src/refract/ExpandVisitor.cc (hunk 2 above)
Comment (pksunkara) [inline src/refract/ExpandVisitor.cc]: That works. Can you please change it so that I can merge this? Thanks.

**Comment by klokane** [comment]
Addressed
